main.tf

```diff
@@ -1,22 +1,36 @@
 locals {
-  enabled    = "${var.enabled == "true" ? true : false }"
-  id         = "${local.enabled ? lower(join(var.delimiter, compact(concat(list(var.namespace, var.stage, var.name), var.attributes)))) : ""}"
-  name       = "${local.enabled ? lower(format("%v", var.name)) : ""}"
-  namespace  = "${local.enabled ? lower(format("%v", var.namespace)) : ""}"
-  stage      = "${local.enabled ? lower(format("%v", var.stage)) : ""}"
-  attributes = "${local.enabled ? lower(format("%v", join(var.delimiter, compact(var.attributes)))) : ""}"
-
-  tags = "${
-      merge( 
-        map(
-          "Name", "${local.id}",
-          "Namespace", "${local.namespace}",
-          "Stage", "${local.stage}"
-        ), var.tags
-      )
-    }"
+  enabled = "${var.enabled == "true" ? true : false }"
+
+  id         = "${lower(join(var.delimiter, compact(concat(list(local.namespace, local.stage, local.name, local.attributes)))))}"
+  name       = "${var.name == "" ? lookup(var.context, "name", "") : lower(format("%v", var.name))}"
+  namespace  = "${var.namespace == "" ? lookup(var.context, "namespace", "") : lower(format("%v", var.namespace))}"
+  stage      = "${var.stage == "" ? lookup(var.context, "stage", "") : lower(format("%v", var.stage))}"
+  attributes = "${length(var.attributes) == 0 ? lookup(var.context, "attributes", "") : lower(format("%v", join(var.delimiter, compact(var.attributes))))}"
+
+  generated_tags = {
+    "Name"      = "${local.enabled ? local.id : ""}"
+    "Namespace" = "${local.enabled ? local.namespace : ""}"
+    "Stage"     = "${local.enabled ? local.stage : ""}"
+  }
+
+  tags = "${merge(local.generated_tags, var.tags)}"
```

Comment:
Need to incorporate the `var.context["tags"]` here some how

Reply:
it doesn't seem possible :(